lua/nvim-treesitter/indent.lua

```diff
@@ -12,21 +25,17 @@ local function get_last_node_at_line(root, lnum)
   return root:descendant_for_range(lnum - 1, col, lnum - 1, col)
 end
 
-local function get_matching_prev_sibling(anchor, start, matcher)
-  local start_row, start_col = start[1], start[2]
-  local node = anchor:descendant_for_range(start_row, start_col, start_row, start_col)
-  local pos = 1
-  -- TODO: reconsider this 999 limit or do something differently in future.
-  --       if anchor has more than 999 children, this would not work.
-  while pos < 999 and node and not matcher(node) do
-    node = node:prev_sibling()
-    pos = pos + 1
+local function find_delimiter(bufnr, node, delimiter)
```

Comment:
this is actually a flaw that we don't use the whole pattern but only the capture. Because of this we need to find the child manually here which is not nice.